core/src/main/resources/hudson/model/Messages.properties

```diff
@@ -195,6 +188,11 @@ Job.you_must_use_the_save_button_if_you_wish=You must use the Save button if you
 Label.GroupOf=group of {0}
 Label.InvalidLabel=invalid label
 Label.ProvisionedFrom=Provisioned from {0}
+LabelExpression.InvalidBooleanExpression=Invalid boolean expression: {0}
+LabelExpression.LabelLink=<a href="{0}{2}">Label {1}</a> is serviced by {3,choice,0#no nodes|1#1 node|1<{3} nodes}{4,choice,0#|1# and 1 cloud|1< and {4} clouds}. \
+  Permissions or other restrictions provided by plugins may prevent this job from running on those nodes.
+LabelExpression.NoMatch=There''s no agent/cloud that matches this assignment
```

Comment:
Sure. But it feels like that warrants a separate PR specifically to deal with apostrophes in messages - it seems unlikely that this is the only case. That PR could then deal with the desired approach for Unicodifying punctuation (e.g using "proper" quotation marks too, if they appear in messages).

Reply:
Yes, just suggesting it here since you are touching the line of code anyway.

Reply:
Is prose preferred over terseness btw? This specific message sounds like "No agent/cloud matches this label expression" would be fine (also dropping the "assignment" term, which I think is outdated and/or not applicable in all contexts.

Reply:
> No agent/cloud matches this label expression

Reply:
Done. I also reworded the other label validation messages to better match each other, and to avoid mentioning "jobs" when the validation may well be happening completely outside of any job context.